domain/watcher.go

```diff
@@ -75,8 +75,36 @@ func (f *WatcherFactory) NewNamespaceMapperWatcher(
 	), nil
 }
 
+// NewFilterWatcher returns a watcher that combines multiple value
+// watchers into a single watcher.
+func (f *WatcherFactory) NewFilterWatcher(
+	filterOpts ...eventsource.FilterOption,
+) (watcher.NotifyWatcher, error) {
+	base, err := f.newBaseWatcher()
+	if err != nil {
+		return nil, errors.Annotate(err, "creating base watcher")
+	}
+
+	return eventsource.NewFilterWatcher(base, filterOpts...), nil
+}
+
+// NewFilterWatcher returns a watcher that combines multiple value
+// watchers into a single watcher.
```

Comment:
Same as above. The comment should also not be the same as the above - at the least it should briefly mention what the mapper arg is for.